server/komodo-rest/src/main/java/org/komodo/servicecatalog/datasources/MongoDBDefinition.java

```diff
@@ -98,23 +98,17 @@ public Properties getDataSourceProperties(DefaultServiceCatalogDataSource source
 
     @Override
     public Properties getWFSDataSourceProperties(DefaultServiceCatalogDataSource scd, String jndiName) {
-        Properties props = new Properties();
-
-        // consult Teiid documents for all the properties; Then map to properties from OpenShift Service
-        props.setProperty("swarm.resource-adapter.resource-adapters." + scd.getName() + ".module=",
-                "org.jboss.teiid.resource-adapter.mongodb");
-
-        ds(props, scd, "class-name", "org.teiid.resource.adapter.mongodb.MongoDBManagedConnectionFactory");
-        ds(props, scd, "jndi-name", jndiName);
-        ds(props, scd, "enabled", "true");
-        ds(props, scd, "use-java-context", "true");
+        Properties props = setupResourceAdapter(scd.getName(), "org.jboss.teiid.resource-adapter.mongodb",
+                "org.teiid.resource.adapter.mongodb.MongoDBManagedConnectionFactory", jndiName);
         ds(props, scd, "SecurityType", "SCRAM_SHA_1");
         ds(props, scd, "Ssl", "false");
         ds(props, scd, "AuthDatabase", scd.canonicalEnvKey("database_name"));
         ds(props, scd, "RemoteServerList", scd.canonicalEnvKey("DATABASE_SERVICE_NAME")+":27017");
         ds(props, scd, "Database", scd.canonicalEnvKey("database_name"));
-        ds(props, scd, "Username", scd.canonicalEnvKey("database-user"));
-        ds(props, scd, "Password", scd.canonicalEnvKey("database-password"));
+        ds(props, scd, "Username", scd.canonicalEnvKey("username"));
+        ds(props, scd, "Password", scd.canonicalEnvKey("password"));
```

Comment:
Are these not in danger of falling foul of one mongodb implementation despite working with another? Might it be safer to simply apply both sets thereby covering all bases since its difficult to know which are preferred?

Reply:
no, they were placeholders before, same with other sources even now. I do not think Mark ever did the publish operation on MongoDB. Thus did not get to this initially. (Actually, I did not do in OpenShift either because of state of the Beetle application in repo currently)